plugins/auto-sizes/hooks.php

```diff
@@ -156,6 +156,15 @@ function auto_sizes_improve_image_sizes_attributes( string $content ): string {
 		return $content;
 	}
 
+	// Bail early if the responsive images are disabled.
+	if ( null === $processor->get_attribute( 'sizes' ) ) {
+		$processor->remove_attribute( 'data-needs-sizes-update' );
+		$processor->remove_attribute( 'data-align' );
+		$processor->remove_attribute( 'data-resize-width' );
```

Comment:
Minor point, but there is some code duplication here. You could refactor this a bit to not have to maintain two sets of attributes to remove:

```suggestion
	$remove_data_attributes = static function () use ( $processor ): void {
		$processor->remove_attribute( 'data-needs-sizes-update' );
		$processor->remove_attribute( 'data-align' );
		$processor->remove_attribute( 'data-resize-width' );
	};

	// Bail early if the responsive images are disabled.
	if ( null === $processor->get_attribute( 'sizes' ) ) {
		$remove_data_attributes();
```

And then `$remove_data_attributes` could be reused below:

https://github.com/WordPress/performance/blob/e83ac2d898300ea3b987a75db48b62c6b5423561/plugins/auto-sizes/hooks.php#L218-L223